### ISSUE TITLE
no-string-literal conflicts with string indexed type

### ISSUE THREAD

**Issue by Jameskmonger** [palantir/tslint#1419]
### Bug Report

- __TSLint version__: `3.13.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
headers: { [key: string]: string; };

req.headers['origin'];
```

with `tslint.json` configuration:

```json
"no-string-literal": true
```

#### Actual behavior

Using a string literal for `req.headers['origin']` fails with tslint, not using a string literal (`req.headers.origin`) causes the build to fail as the object must be string-indexed.

#### Expected behavior

I am unsure how tslint interacts with the typings. However the ideal solution would be for tslint to understand that `headers` is string-indexed and so not apply the rule.
